portfolio/src/main/webapp/script.js

```diff
@@ -13,21 +13,89 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * A wrapper around slowFill(), with the specific purpose of slowly
+ * filling out the biography section of the website.
  */
-function addRandomFact() {
-  const facts =
-      [
-        'I am 19 years old!',
-        'I once ran arch linux on Windows using WSL for the lolz!',
-        'My dogs\' names are Polly and Rex!',
-        'My \'to-read\' list is currently growing at a rate of about 2bpm (books per month)!',
-      ];
+const slowFillBiography = async () => {
+  const biography_text = 
+      'Hi! My name\'s Thomas, and I\'m just about to start my third year of ' +
+      'studies in University College Dublin. I\'m currently a STEP intern ' +
+      'at Google, and this portfolio aims to showcase my contributions ' + 
+      'during my time here.';
+  const output_element = document.getElementById('biography');
+  const wait_function = () => (Math.floor(Math.random() * 50) + 50);
+  slowFill(output_element, biography_text, wait_function);
+}
 
-  // Pick a random greeting.
-  const fact = facts[Math.floor(Math.random() * facts.length)];
+/**
+ * Fill the text field of a given entity with a given string of text, character
+ * by character, at a programmable rate.
+ * @param entity The entity to write to the text field of.
+ * @param text The text to write to the text field of the entity.
+ * @param wait_function A function returning an integer value which is the
+ *    number of milliseconds for which the site will wait between printing
+ *    successive tokens.
+ */
+const slowFill = async (entity, text, wait_function) => {
+  /*
+    Boolean value indicating if there is a cursor attached to the
+    end of the text field being written to.
+  */
+  let cursor_attached = false;
 
-  // Add it to the page.
-  const factContainer = document.getElementById('fact-container');
-  factContainer.innerText = fact;
+  /* 
+    wait_function must be non-null.
+    all null inputs are normalised to a known default.
+   */
+  if (wait_function == null) wait_function = function() {0};
+
+  const removeCursor = () => {
+    if(cursor_attached){
+      entity.innerText = 
+          entity.
+              innerText.
+                  substr(0, entity.innerText.length - 1);
+      cursor_attached = !cursor_attached;
+    }
+  };
+  const attachCursor = () => {
+    entity.innerText += cursor_attached ? '' : '|';
+    cursor_attached = !cursor_attached;
+  };
+  
+  for(let word of text.split(' ')){
```

Comment:
I am not clear what is the goal here. Comment with the necessary details? Or attach a screenshot of the goal?

Reply:
I've improved the documentation now 👍 Let me know if it needs more work https://github.com/thomasogara/step-repository/pull/2/commits/1411993e99833c3eac58a7a59506bd1cef8ed5d0

Reply:
I also added documentation for the nested for loops, just to better explain the intent of the function 👍 https://github.com/thomasogara/step-repository/pull/2/commits/d9e27cc5b6d81a0b38efc9feb1f7be6f610df1db